NPE in ShareIntentReceiverActivity

```
java.lang.IllegalStateException: Could not execute method of the activity
at android.view.View$1.onClick(View.java:3633)
at android.view.View.performClick(View.java:4240)
at android.view.View$PerformClick.run(View.java:17721)
at android.os.Handler.handleCallback(Handler.java:730)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:5103)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:525)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:525)
at android.view.View$1.onClick(View.java:3628)
... 11 more
Caused by: java.lang.NullPointerException
at org.wordpress.android.ui.ShareIntentReceiverActivity.savePreferences(ShareIntentReceiverActivity.java:284)
at org.wordpress.android.ui.ShareIntentReceiverActivity.startActivityAndFinish(ShareIntentReceiverActivity.java:205)
at org.wordpress.android.ui.ShareIntentReceiverActivity.shareIt(ShareIntentReceiverActivity.java:224)
at org.wordpress.android.ui.ShareIntentReceiverActivity.onShareClicked(ShareIntentReceiverActivity.java:212)
```

--- a/src/org/wordpress/android/ui/ShareIntentReceiverActivity.java
+++ b/src/org/wordpress/android/ui/ShareIntentReceiverActivity.java
@@ -95,8 +95,8 @@ public class ShareIntentReceiverActivity extends SherlockFragmentActivity implem
                 startActivityAndFinish(new Intent(this, EditPostActivity.class));
             }
         } else {
-            String[] actions = new String[]{getString(R.string.share_action_post),
-                    getString(R.string.share_action_media)};
+            String[] actions = new String[]{getString(R.string.share_action_post), getString(
+                    R.string.share_action_media)};
             ArrayAdapter<String> actionAdapter = new ArrayAdapter<String>(themedContext,
                     R.layout.sherlock_spinner_dropdown_item, actions);
             mActionSpinner.setAdapter(actionAdapter);
@@ -124,8 +124,7 @@ public class ShareIntentReceiverActivity extends SherlockFragmentActivity implem
             startActivity(new Intent(this, WelcomeActivity.class));
             finish();
         } else {
-            ToastUtils.showToast(getBaseContext(), R.string.cant_share_no_visible_blog,
-                    ToastUtils.Duration.LONG);
+            ToastUtils.showToast(getBaseContext(), R.string.cant_share_no_visible_blog, ToastUtils.Duration.LONG);
             finish();
         }
     }
@@ -241,8 +240,7 @@ public class ShareIntentReceiverActivity extends SherlockFragmentActivity implem
                 shareIt();
                 return true;
             } else { // blog is hidden or has been deleted, reset settings
-                SharedPreferences.Editor editor = PreferenceManager
-                        .getDefaultSharedPreferences(this).edit();
+                SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(this).edit();
                 editor.remove(SHARE_TEXT_BLOG_ID_KEY);
                 editor.commit();
                 ToastUtils.showToast(this, R.string.auto_sharing_preference_reset_caused_by_error,
@@ -262,8 +260,7 @@ public class ShareIntentReceiverActivity extends SherlockFragmentActivity implem
                 shareIt();
                 return true;
             } else { // blog is hidden or has been deleted, reset settings
-                SharedPreferences.Editor editor = PreferenceManager
-                        .getDefaultSharedPreferences(this).edit();
+                SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(this).edit();
                 editor.remove(SHARE_IMAGE_BLOG_ID_KEY);
                 editor.remove(SHARE_IMAGE_ADDTO_KEY);
                 editor.commit();
@@ -275,9 +272,12 @@ public class ShareIntentReceiverActivity extends SherlockFragmentActivity implem
     }
 
     private void savePreferences() {
+        // If current blog is not set don't save preferences
+        if (WordPress.currentBlog == null) {
+            return ;
+        }
         if (mAlwaysUseCheckBox.isChecked()) {
-            SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(this)
-                    .edit();
+            SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(this).edit();
             if (isSharingText()) {
                 editor.putInt(SHARE_TEXT_BLOG_ID_KEY, WordPress.currentBlog.getLocalTableBlogId());
             } else {

